app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -476,9 +476,7 @@ def save_vm(uuid, attr)
       if attr.has_key?(:volumes_attributes)
         vm.volumes.each do |vm_volume|
           volume_attrs = attr[:volumes_attributes].values.detect {|vol| vol[:id] == vm_volume.id}
-          if volume_attrs.class == Hash && volume_attrs.key?(:size_gb)
-            vm_volume.size_gb = volume_attrs[:size_gb]
-          end
+          vm_volume.size_gb = volume_attrs[:size_gb] if volume_attrs[:size_gb].present?
```

Comment:
Perhaps it should also check if it's not nil? I assume that why it checked for a hash

Reply:
`if volume_attrs[:size_gb].present?` is good enough as it only returns true if  `volume_attrs[:size_gb]` has a value... 

however I just added a guard against volume_attrs being `blank` just incase.